tests/test_object_detection.py

```diff
@@ -0,0 +1,523 @@
+            for p in prediction
+        ]
+    return np.array(prediction, dtype=object)
+
+
+def generate_annotations(num_annotations, num_classes=5, max_boxes=5, image_size=300):
+    """Generates num_annotations number of annotations based on passed in hyperparameters in same format as expected by find_label_issues and get_label_quality_scores"""
+
+    annotations = []
+    for i in range(num_annotations):
+        annotations.append(generate_annotation(num_classes, image_size, max_boxes))
+    return annotations
+
+
+def generate_annotation(num_classes, image_size, max_boxes):
+    """Generates a single annotation based on passed in hyperparameters in same format as expected by find_label_issues and get_label_quality_scores"""
+
+    num_boxes = np.random.randint(low=1, high=max_boxes)
+    bboxes = np.array([generate_bbox(image_size) for _ in range(num_boxes)])
+    labels = generate_labels(num_classes, num_boxes)
+    annotation = {"bboxes": bboxes, "labels": labels}
+    return annotation
+
+
+def generate_labels(num_classes, num_boxes):
+    """Generates num_boxes number of labels with possible values [0-num_classes)"""
+    return np.random.choice(num_classes, num_boxes)
+
+
+def generate_bbox(image_size):
+    """Generates a single bounding box x1,y1,x2,y2 with coordinates lower than image_size"""
+    x2 = np.random.randint(low=2, high=image_size - 1)
+    y2 = np.random.randint(low=2, high=image_size - 1)
+    x_shift = np.random.randint(low=1, high=x2)
+    y_shift = np.random.randint(low=1, high=y2)
+    x1 = x2 - x_shift
+    y1 = y2 - y_shift
+    return [x1, y1, x2, y2]
+
+
+warnings.filterwarnings("ignore")
+
+good_labels = generate_annotations(5, num_classes=10, max_boxes=10)
+good_predictions = generate_predictions(
+    5, good_labels, num_classes=10, max_boxes=12, is_issue=False
+)
+
+bad_labels = generate_annotations(5, num_classes=10, max_boxes=10)
+bad_predictions = generate_predictions(5, bad_labels, num_classes=10, max_boxes=12, is_issue=True)
+
+labels = good_labels + bad_labels  # 10 labels
+predictions = (
+    good_predictions + bad_predictions
+)  # 10 predictions, [:5] is perfect predictions, [5:] is bad predictions
+
+
+def test_get_label_quality_scores():
+    scores = get_label_quality_scores(labels, predictions)
+    assert len(scores) == len(labels)
+    assert (scores <= 1.0).all()
+    assert len(scores.shape) == 1
+    assert (scores[:5] > 0.9).all()  # perfect annotations get high scores
+    assert (scores[5:] < 0.7).all()  # label issues get low scores
+
+
+@pytest.mark.parametrize(
+    "agg_weights",
+    [
+        {"overlooked": 1.0, "swap": 0.0, "badloc": 0.0},
+        {"overlooked": 0.0, "swap": 1.0, "badloc": 0.0},
+        {"overlooked": 0.0, "swap": 0.0, "badloc": 1.0},
+    ],
+)
+def test_get_label_quality_scores_custom_weights(agg_weights):
+    scores = get_label_quality_scores(labels, predictions, aggregation_weights=agg_weights)
+    if agg_weights["swap"] == 1.0:
```

Comment:
Curious case of swap: 
when `{"overlooked": 0.0, "swap": 1.0, "badloc": 0.0}`
```
scores = [0.82794274, 0.9877464 , 0.97805729, 0.92383053, 0.99535796,
       0.49723733, 0.65475014, 0.5852968 , 0.62312097, 0.95911965]
```
when `{"overlooked": 1.0, "swap": 0.0, "badloc": 0.0}`
```
scores = [1.00000000e+00, 1.00000000e+00, 1.00000000e+00, 1.00000000e+00,
       1.00000000e+00, 5.30818008e-03, 2.58525436e-05, 2.96260415e-03,
       3.59268220e-07, 1.26750207e-09]
```

when `{"overlooked": 0.0, "swap": 0.0, "badloc": 1.0}`
```
[1.00000000e+00, 1.00000000e+00, 1.00000000e+00, 1.00000000e+00,
       1.00000000e+00, 4.51579879e-03, 2.00049667e-05, 3.93848864e-03,
       1.29199511e-06, 3.01253446e-09]
```